packages/availity-reactstrap-validation/src/AvStore.js

```diff
@@ -0,0 +1,364 @@
+      this.updateInputValidations({}, name);
+      // remove inputs last because watchers is needed for validations
+      delete this.inputs[name];
+    };
+  }
+
+  notifySubscriber = (inputId) => {
+    // TODO: determine what information to provide on notification
+    this.subscribers[inputId]();
+  }
+
+  updateValues = (values) => {
+    this.values = values;
+    Object.keys(this.inputs).forEach(inputId => {
+      const newValue = this.getDefaultValue(inputId);
+      const currentValue = this.getValue(inputId);
+      if (newValue !== currentValue) {
+        this.updateInput(inputId, newValue, true);
+      }
+    });
+  }
+
+  updateValue = (inputId, value, validate) => {
+    this.updateInput(inputId, {value}, validate);
+  }
+
+  updateInput = (inputId, input, validate) => {
+    Object.assign(this.inputs[inputId], input);
+    if (validate) {
+      this.onEvent(inputId, 'onChange');
+    }
+  }
+
+  // grabs errors but adds in error messages as needed
+  getErrors = (inputId) => {
+    const errors = (this.getInput(inputId) || {}).error;
+    return Object.keys(errors).reduce((output, errorKey) => {
+      const error = errors[errorKey];
+      if (!error) {
+        return output;
+      }
+      // if validator returned string, return as is
+      if (isString(error)) {
+        output[errorKey] = error;
+        return output;
+      }
+      // check if input has error message defined
+      const inputErrorMessage = _get(this.allInputValidations, [inputId, errorKey, 'errorMessage']);
+      if (inputErrorMessage) {
+        output[errorKey] = inputErrorMessage;
+        return output;
+      }
+      // TODO: Get default error messages
+      return output;
+    }, {});
+  }
+
+  getDefaultValue = (inputId) => _get(this.values, inputId);
+  getInput = (inputId) => _get(this.inputs, inputId);
+  getValue = (inputId) => (this.getInput(inputId) || {}).value;
+
+  getValues() {
+    return Object.keys(this.inputs).reduce((output, inputId) => {
+      _set(output, inputId, this.getValue(inputId) );
+      return output;
+    }, {});
+  }
+
+
+  /*
+    input event types:
+    'onInput', 'onChange', 'onBlur', 'onFocus' - input events (have inputs just update state and call onInputChange?)
+    inputChange - when an input object is changed, has input id changed
+    validatorChange - when a validation rule changes, re-run validaitons using this rule
+    errorMessageChange - if a default error message changes, so that inputs using default messages can update
+    onSubmit - submit event
+
+    events that are sent to inputs:
+    inputChange, errorMessageChange
+
+    event = eventType or {
+      type: eventType
+      event specific values (ex. input, validator, errorKey)
+    }
+  */
+  onEvent(inputId, event) {
+    const runningValidators = [
+      this.validateOne(inputId, event),
+    ];
+    if (['onChange', 'onInput'].indexOf(event) >= 0 && this.watchers[inputId]) {
```

Comment:
Cache this array so that it is not created every time this method runs. It can also be used in the other places the same array is created.

Reply:
I just moved ['onChange', 'onInput'] to an array to be re-used. Or did you mean to cache what validators should run for each input/event combination?

Reply:
I meant the `['onChange', 'onInput']`.